deepchem/models/torch_models/layers.py

```diff
@@ -2682,3 +2682,336 @@ def forward(
+      self.b_AP: torch.Tensor = torch.zeros((self.n_hidden_AP,))
+      self.AP_bn = nn.BatchNorm1d(num_features=self.n_hidden_AP,
+                                  eps=1e-3,
+                                  momentum=0.99,
+                                  affine=True,
+                                  track_running_stats=True)
+
+      self.W_PP: torch.Tensor = init(
+          torch.empty(self.n_pair_input_feat, self.n_hidden_PP))
+      self.b_PP: torch.Tensor = torch.zeros((self.n_hidden_PP,))
+      self.PP_bn = nn.BatchNorm1d(num_features=self.n_hidden_PP,
+                                  eps=1e-3,
+                                  momentum=0.99,
+                                  affine=True,
+                                  track_running_stats=True)
+
+      self.W_P: torch.Tensor = init(
+          torch.empty(self.n_hidden_P, self.n_pair_output_feat))
+      self.b_P: torch.Tensor = torch.zeros((self.n_pair_output_feat,))
+      self.P_bn = nn.BatchNorm1d(num_features=self.n_pair_output_feat,
+                                 eps=1e-3,
+                                 momentum=0.99,
+                                 affine=True,
+                                 track_running_stats=True)
+    self.built = True
+
+  def __repr__(self) -> str:
+    """
+    Returns a string representation of the object.
+
+    Returns:
+    -------
+    str: A string that contains the class name followed by the values of its instance variable.
+    """
+    return (
+        f'{self.__class__.__name__}(n_atom_input_feat:{self.n_atom_input_feat},n_pair_input_feat:{self.n_pair_input_feat},n_atom_output_feat:{self.n_atom_output_feat},n_pair_output_feat:{self.n_pair_output_feat},n_hidden_AA:{self.n_hidden_AA},n_hidden_PA:{self.n_hidden_PA},n_hidden_AP:{self.n_hidden_AP},n_hidden_PP:{self.n_hidden_PP},batch_normalize:{self.batch_normalize},update_pair:{self.update_pair},init:{self.init},activation:{self.activation})'
+    )
+
+  def forward(
+      self, inputs: List[Union[np.ndarray, np.ndarray, np.ndarray, np.ndarray]]
+  ) -> List[Union[torch.Tensor, torch.Tensor]]:
+    """
+    Creates weave tensors.
+
+    Parameters
+    ----------
+    inputs: List[Union[np.ndarray, np.ndarray, np.ndarray, np.ndarray]]
+      Should contain 4 tensors [atom_features, pair_features, pair_split,
+      atom_to_pair]
+
+    Returns:
+    -------
+    List[Union[torch.Tensor, torch.Tensor]]
+      A: Atom features tensor with shape[total_num_atoms,atom feature size]
+      P: Pair features tensor with shape[total num of pairs,bond feature size]
+    """
+    #Converting the input to torch tensors
+    atom_features: torch.Tensor = torch.tensor(inputs[0])
+    pair_features: torch.Tensor = torch.tensor(inputs[1])
+
+    pair_split: torch.Tensor = torch.tensor(inputs[2])
+    atom_to_pair: torch.Tensor = torch.tensor(inputs[3])
+
+    activation = self.activation_fn
+
+    AA: torch.Tensor = torch.matmul(atom_features.type(torch.float32),
```

Comment:
comment the shapes of the tensors after each operation

Reply:
Yes, added shape